ldap/servers/slapd/ssl.c

```diff
@@ -2659,6 +2636,9 @@ slapd_extract_cert(Slapi_Entry *entry, int isCA)
         }
     }
     rv = SECSuccess;
+
+    cert_extract_file = slapi_ch_strdup(certfile);
```

Comment:
slapd_extract_cert is called several times. For example for 'cn=encryption,cn=config' and for each of its children. Should not  you call slapi_ch_free(&cert_extract_file) before setting it.